src/java.base/share/classes/java/security/Security.java

```diff
@@ -192,6 +199,22 @@ private static void initialize() {
             }
         }
 
+        // check if CRIU checkpoint mode is enabled, if it is - reconfigure the security providers
+        boolean isCRIUCapable = Boolean.valueOf(System.getProperty("org.eclipse.openj9.criu.isCRIUCapable", "false"));
+        if (isCRIUCapable) {
+            try {
+                // Invoke the isCRIUSupportEnabled method from the CRIUSupport class
+                Class runnable = Class.forName("org.eclipse.openj9.criu.CRIUSupport", true, ClassLoader.getSystemClassLoader());
+                Method criuMethod =  runnable.getMethod("isCRIUSupportEnabled", new Class[] {});
```

Comment:
use `InternalCRIUSupport.isCRIUSupportEnabled()` to avoid reflection.

Reply:
I get the error `j9criu.2      *   ** ASSERTION FAILED ** at /home/jenkins/workspace/Build_JDK11_x86-64_linux_criu_Personal/openj9/runtime/criusupport/criusupport.cpp:52: ((criuJVMCheckpointExceptionClass) != NULL)`

Reply:
changed